apollo-runtime/src/main/java/com/apollographql/android/cache/HttpCacheInterceptor.java

```diff
@@ -54,9 +68,15 @@ private boolean shouldSkipNetwork(Request request) {
 
   private Response cacheOnlyResponse(Request request) throws IOException {
     String cacheKey = request.header(CACHE_KEY_HEADER);
-    Response cachedResponse = cache.read(cacheKey);
-    if (cachedResponse != null) {
-      return cachedResponse;
+    Response cacheResponse = cache.read(cacheKey);
+    if (cacheResponse != null) {
+      if (cache.isStale(cacheResponse)) {
+        cache.remove(cacheKey);
```

Comment:
A little concerned about the CacheInterceptor including eviction logic. I feel the CacheInterceptor should only use expiration for *fetching preference*.

Staleness is two-fold indicator - one, it may be used to signal a change in "fetch-strategy", i.e. if the cache is stale, prefer the network. But, in that situation, you may want to still return cache if network fails. If a particular cache thinks keeping stale data around is harmful, it should return `null` on a `read` of stale data, or even run it's own eviction subroutine.